parser/test/parser_test.c

```diff
@@ -1,21 +1,96 @@
+                      "              {\n"
+                      "                \"type\": \"AttributeReference\",\n"
+                      "                \"attribute_name\": \"a\"\n"
+                      "              }\n"
+                      "            ]\n"
+                      "          },\n"
+                      "          {\n"
+                      "            \"type\": \"AttributeReference\",\n"
+                      "            \"attribute_name\": \"b\"\n"
+                      "          }\n"
+                      "        ],\n"
+                      "        \"from\": [\n"
+                      "          {\n"
+                      "            \"type\": \"ParseSimpleTableReference\",\n"
+                      "            \"table_name\": \"integer_table\"\n"
+                      "          }\n"
+                      "        ],\n"
+                      "        \"group_by\": [\n"
+                      "          {\n"
+                      "            \"type\": \"AttributeReference\",\n"
+                      "            \"attribute_name\": \"b\"\n"
+                      "          }\n"
+                      "        ]\n"
+                      "      }\n"
+                      "    ]\n"
+                      "  }\n"
+                      "}";
 
-    char *json_output = json_stringify(node);
-    printf("%s", json_output);
-    free(json_output);
+    char *query2 = "SELECT CommonPrefix(ip) FROM ip_table;";
+    char *expected2 = "{\n"
+                      "  \"type\": \"ParseStatementSetOperation\",\n"
+                      "  \"set_operation_query\": {\n"
+                      "    \"type\": \"ParseSetOperation\",\n"
+                      "    \"set_operation_type\": \"Select\",\n"
+                      "    \"operands\": [\n"
+                      "      {\n"
+                      "        \"type\": \"ParseSelect\",\n"
+                      "        \"select\": [\n"
+                      "          {\n"
+                      "            \"type\": \"FunctionCall\",\n"
+                      "            \"name\": \"CommonPrefix\",\n"
+                      "            \"arguments\": [\n"
+                      "              {\n"
+                      "                \"type\": \"AttributeReference\",\n"
+                      "                \"attribute_name\": \"ip\"\n"
+                      "              }\n"
+                      "            ]\n"
+                      "          }\n"
+                      "        ],\n"
+                      "        \"from\": [\n"
+                      "          {\n"
+                      "            \"type\": \"ParseSimpleTableReference\",\n"
+                      "            \"table_name\": \"ip_table\"\n"
+                      "          }\n"
+                      "        ]\n"
+                      "      }\n"
+                      "    ]\n"
+                      "  }\n"
+                      "}";
```

Comment:
I know this is really ugly, but it's a temporary solution. We'll need to talk about testing frameworks.

Reply:
It is ok for now I think, but yes we should create a frameworks to test the output of the parser.
Also we should use testing frameworks like gtest